docs/architecture/application/cli.md

```diff
@@ -570,49 +612,105 @@ product_blocks:
-life cycle states the resource type will be optional. If `required` is not
-specified the resource type will always be optional. If the resource type is
-optional in one of its lifecycle states then the `default` attribute specifies
-an optional default value.  The `modifiable` flag is to indicate if this
-resource type can be modified in a modify workflow. And the `description` is
-used in the created migration.
+Resource types are described by the following:
+
+- `name`: name of the resource type, usually in snake case
+- `decription`: resource type description
+- `required`: if the resource type is required starting from lifecycle state `inital`, `provisioning`
+              or `active`, when omitted the resource type will always be optional
+- `modifiable`: indicate if the resource type can be altered by a modify workflow, when omitted
+                no code will be generated to modify the resource type, currently only supported for simple types
+- `default`: specify the default for this resource type, this is mandatory when `required` is set to
+             `intitial`, will be `None` if not specified
+- `validations`: specify the validation `id` and `description`, generates a skeleton
+                 validation function used in a new annotated type
+- `type`: see explanation below
+
+#### resource type types
+
+The following types are supported when specifying resource types:
+
+- primitive types like `int`, `str`, and `bool`
+  ```yaml
+  - name: circuit_id
+    type: int
+  ```
+- types with a period in there name will generate code to import that type, e.q. `ipaddress.IPv4Address` 
+  ```yaml
+  - name: ipv4_loopback
+    type: ipaddress.IPv4Address
+  ```
+- existing product block types like `UserGroup`, possible types will be read from `products/product_blocks`
+  ```yaml
+  - name: group
+    type: UserGroup
+  ```
+- the type `list`, used together with `min_items` and `max_items` to specify the constraints, and
+  `list_type` to specify the type of the list items
+  ```yaml
+  - name: link_members
+    type: list
+    list_type: Link
+    min_items: 2
+    max_items: 2
+  ```
+- validation code for constrained a `int` is generated when `min_value` and `max_value` are specified
+  ```yaml
+    - name: ims_id
+      type: int
+      min_value: 1
+      max_value: 32_767
+  ```
+
+#### workflows
+
+```yaml
+  - name: create
+    validations:
+      - id: endpoints_cannot_be_on_same_node
+        description: "Service endpoints must land on different nodes"
+  - name: validate
+    enabled: false
+    validations:
+      - id: validate_ims_administration
+        description: "Validate that the node is correctly administered in IMS"
+```
+
+The following optional workflow configuration is supported:
+
+- `name`: can be either `create`, `modify`, `terminate` or `validate`
+- `enables`: to enable or disable the generation of code for that type of workflow, is `true` when omitted
```

Comment:
enabled instead of enables?

Reply:
Good catch, this should be `enabled` of course.